### PR TITLE
Only display approved haikus publicly

### PR DESCRIPTION
**Pull request by lukemcfarlane**

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -2,4 +2,10 @@
 
 class Submission < ApplicationRecord
   validates :body, presence: true
+
+  enum status: {
```
Comment (Aupajo) [inline app/models/submission.rb]: `enum` 😍

File: spec/support/haiku_helper.rb [hunk 2]
```diff
@@ -8,6 +8,16 @@ def submit_haiku(poem = 'A haiku', by: nil)
     fill_in 'Your haiku', with: poem
     click_on 'Send'
   end
+
+  def approve_haiku(poem, by: nil)
+    # TEMP this should eventually use admin approve view:
+    submissions = Submission.where(body: poem)
+    submissions = submissions.where(submitted_by: by) if by.present?
+
+    return unless submissions.one?
```
Comment (Aupajo) [inline spec/support/haiku_helper.rb]: Should this raise an error? If I wrote this test:

```ruby
submit_haiku('A haiku!')
submit_haiku('A haiku!')
approve_haiku('A haiku!')
```

I would not expect _neither_ haiku to be approved.

How about…

```suggestion
fail("Could not find one haiku matching approval criteria") unless submissions.one?
```

File: spec/support/haiku_helper.rb [hunk 3]
```diff
@@ -8,6 +8,16 @@ def submit_haiku(poem = 'A haiku', by: nil)
...
+
+    submissions.first.update!(status: Submission.statuses[:approved])
```
Comment (Aupajo) [inline spec/support/haiku_helper.rb]: `enum` should let you do…

```suggestion
submissions.first.approved!
```

**Comment by lukemcfarlane** [comment]
Those were excellent suggestions, thanks @Aupajo! Can you please re-review?

**Pull request merged**